### ISSUE TITLE
Summarize model generalization performance

### ISSUE THREAD

**Issue by kschelonka** [AllenInstitute/croissant#57]
As a scientist, I want to see whether the model generalizes across different CRE lines. 

- [x] Subset the data into two sets, separating by CRE line, and save on S3
- [x] Update the models so that they can predict on different lines (in this case since it's binary just remove the genotype feature)
- [x] Train the models on one CRE line and evaluate performance by using the other line as the test set (results in 2 models). Can use our existing Fargate infrastructure without any changes, since the only difference are the input data sets.
- [x] Summarize performance metrics
- [x] Make postage-stamp visualizations of ROIs (focus on misclassified ROIs, which are more interesting/important and should be a smaller set)
- [ ] Create a slide for discussion and review

Acceptance criteria:
- [x] Demo slide with summary information from the task.

**Comment by kschelonka** [comment]
Breaking down these tasks further so Taylor and I have a template for what to do:

1. Subset the data into two sets, separating by CRE line, and save on S3.
Best way to do this is probably locally since the data are not large. Can use a jupyter notebook to do it interactively.

    a. Download the train and test data from s3:
URI = "s3://prod.slapp.alleninstitute.org/merged_2line_2project/training_data/2020_07_14-10_21/training_data.jsonl"
    b. Split the train and test data by genotype
    c. Upload the split data to another folder on s3 -- use date prefix
    d. Upload the data to s3, naming with the genotype
    e.  Upload your notebook to s3, in the same "folder"

2.  Update the models so that they can predict on different lines (in this case since it's binary just remove the genotype feature)
    a. Clone the `croissant` repo if haven't already 
    b. Create a new branch with "docker" in the name so that it automatically builds a docker image on push
    c. Make changes required to drop the genotype feature and push
    d. Find the docker image on dockerhub that was created from your push

3. Train the models on one CRE line and evaluate performance by using the other line as the test set (results in 2 models). Can use our existing Fargate infrastructure without any changes, since the only difference are the input data sets.
    a. Submit 2 jobs using the fargate launcher. The training set and test set should be swapped for the two jobs. Specify the docker image created. For instructions, see the README. You will use the `online_training_helper` to get the required environment variables.

4.  Summarize performance metrics
    a. Configure VPN access and log in (further instructions to be given)
    b. Find the runs in the mlflow database, and pull the metrics. You can also find the URIs to the trained models in s3 which can be downloaded.

5.  Make postage-stamp visualizations of ROIs (focus on misclassified ROIs, which are more interesting/important and should be a smaller set)
    a. Download the models and the input data. Again since they are not very large and don't take long to run, can do this locally.
    b. Re-run the models on whatever was used for the "test" data for that model (should be the opposite genotype it was trained on), to get predictions by individual ROI
    c. Create the "postage stamp" from the ROI data, and group them by the prediction type (TP, FP, FN, TN). There is code floating around to do the postage stamp, so we'll go looking for it/ask Dan

6.  Create a slide for discussion and review

**Comment by kschelonka** [comment]
"Postage stamp" code that probably will help: https://github.com/AllenInstitute/segmentation-labeling-app/blob/master/slapp/evaluation/views.py
